src/foundry/client/data/documents/chat-message.d.ts

```diff
@@ -207,52 +180,76 @@ declare global {
 
     /**
      * Render the inner HTML content for ROLL type messages.
-     * @param messageData - The chat message data used to render the message HTML
+     * @param messageData     - The chat message data used to render the message HTML
      * @internal
      */
     protected _renderRollContent(messageData: ChatMessage.MessageData): Promise<void>;
 
+    /** @internal */
     protected override _preCreate(
-      data: ChatMessageDataConstructorData,
+      data: BaseChatMessage.ConstructorData,
       options: DocumentModificationOptions,
-      user: foundry.documents.BaseUser
+      user: BaseUser
     ): Promise<void>;
 
+    /** @internal */
     protected override _onCreate(
-      data: foundry.data.ChatMessageData["_source"],
+      data: BaseChatMessage.Source,
       options: DocumentModificationOptions,
       userId: string
-    ): void;
+    ): Promise<void>;
 
+    /** @internal */
     protected override _onUpdate(
-      data: DeepPartial<foundry.data.ChatMessageData["_source"]>,
+      data: DeepPartial<BaseChatMessage.UpdateData>,
```

Comment:
I don't think we need `DeepPartial` on `UpdateData`. It should already be optional where needed.